src/SelectPanel/SelectPanel.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {useCallback, useMemo} from 'react'
+
+import {FilteredActionList, FilteredActionListProps} from '../FilteredActionList'
+import {OverlayProps} from '../Overlay'
+import {ItemInput} from '../ActionList/List'
+import {FocusZoneSettings} from '../behaviors/focusZone'
+import {DropdownButton, DropdownButtonProps} from '../DropdownMenu'
+import {ItemProps} from '../ActionList'
+import {AnchoredOverlay} from '../AnchoredOverlay'
+import Flex from '../Flex'
+
+export interface SelectPanelProps extends Omit<FilteredActionListProps, 'onChange'> {
+  open: boolean
+  onOpen: (gesture: 'anchor-click' | 'anchor-key-press') => unknown
+  onClose: (gesture: 'click-outside' | 'escape') => unknown
+  renderAnchor?: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+  placeholder?: string
+  selectedItems: ItemInput[]
+  selectionVariant?: 'single' | 'multiple'
```

Comment:
What do you think about making this a [discriminated union type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions)? That’d prevent a component consumer from passing multiple `selectedItems` to a `SelectPanel` with `selectionVariant` `'single'`.

```tsx
type SingleSelectPanelProps = {
  selectionVariant: 'single'
  selectedItems: ItemInput
  onChange?: (selectedItem: ItemInput) => void
}
type MultipleSelectPanelProps = {
  selectedItems: ItemInput[]
  selectionVariant: 'multiple'
  onChange?: (selectedItems: ItemInput[]) => void
}
```

Reply:
Alternatives:
- Get rid of `selectionVariant` and use `selectedItem` vs `selectedItems` (mutually-exclusive)
- Use `selection?: ItemInput | ItemInput[]` instead of `selectedItem`/`selectedItems`